File: sponge-jython/examples/script/py/actions_hello_world.py
"""
Sponge Knowledge base
Hello World action
"""

class HelloWorldAction(Action):
    def onConfigure(self):
        self.withLabel("Hello world").withDescription("Returns a greeting text.")
        self.withArg(StringType("name").withLabel("Your name").withDescription("Type your name."))
        self.withResult(StringType().withLabel("Greeting").withDescription("The greeting text."))
    def onCall(self, name):
        return "Hello World! Hello {}!".format(name)

def onStartup():
    sponge.logger.info("{}", sponge.call("HelloWorldAction", ["Sponge user"]))

File: sponge-jruby/examples/script/rb/actions_hello_world.rb
# Sponge Knowledge base
# Hello World action

class HelloWorldAction < Action
    def onConfigure
        self.withLabel("Hello world").withDescription("Returns a greeting text.")
        self.withArg(StringType.new("name").withLabel("Your name").withDescription("Type your name."))
        self.withResult(StringType.new().withLabel("Greeting").withDescription("The greeting text."))
    end
    def onCall(name)
        return "Hello World! Hello %s!" % [name]
    end
end

def onStartup
    $sponge.logger.info("{}", $sponge.call("HelloWorldAction", ["Sponge user"]))
end
